### PR TITLE
PEGTL Path Parser Extraction

### PR DESCRIPTION
**Pull request by erslavin**
### Description of Change(s)

- Separated out existing PEGTL path parser into its own set of header / source files in preparation for it to be used directly in a PEGTL based text file format parser

### Fixes Issue(s)
-

- [X] I have verified that all unit tests pass with the proposed changes

- [X] I have submitted a signed Contributor License Agreement

### CODE REVIEW

**Comment by jesschimein** [comment]
Filed as internal issue #USD-8973

File: pxr/usd/sdf/pathParser.cpp [hunk 1]
```diff
@@ -0,0 +1,212 @@
+//
+// Copyright 2016 Pixar
```
Comment (crevilla) [inline pxr/usd/sdf/pathParser.cpp]: Need to update the year in the copyright since this is a new file

File: pxr/usd/sdf/pathParser.h [hunk 2]
```diff
@@ -0,0 +1,182 @@
+//
+// Copyright 2016 Pixar
```
Comment (crevilla) [inline pxr/usd/sdf/pathParser.h]: Same here with the copyright year update

File: pxr/usd/sdf/pathParser.h [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+PXR_NAMESPACE_OPEN_SCOPE
+
+bool
+Sdf_ParsePath(std::string const &pathStr, SdfPath *path, std::string *errMsg);
+
+namespace Sdf_PathParser {
+
+////////////////////////////////////////////////////////////////////////
+// SdfPath grammar:
+
```
Comment (crevilla) [inline pxr/usd/sdf/pathParser.h:44]: With this being in a header, I'm glad to see the use of the explicit namespace of tao::TAO_PEGTL_NAMESPACE.
But it's a bit verbose and hard to read. I wondering if a `using PEGTL = tao::TAO_PEGTL_NAMESPACE;` here (or PEGTL_NS) would give us the best of both worlds yielding a shorter but still explicit namespace for all the declarations.

Comment (erslavin) [reply pxr/usd/sdf/pathParser.h:44]: shortend namespace with `namespace PEGTL_NS = tao::TAO_PEGTL_NAMESPACE;`

**Pull request merged**